streams/resources/rs-utils.js

```diff
@@ -1,5 +1,44 @@
 'use strict';
 (function () {
+  // Fake setInterval-like functionality in environments that don't have it
+  class IntervalHandle {
+    constructor(callback, delayMs) {
+      this.callback = callback;
+      this.delayMs = delayMs;
+      this.endEpochMs = Date.now() + delayMs;
+      this.promise = Promise.resolve();
+      this.cancelled = false;
+      this.promise.then(() => this.check());
+    }
+
+    check() {
+      if ((this.endEpochMs - Date.now()) <= 0) {
+        this.callback();
+        if (this.cancelled)
+          return;
+        this.endEpochMs = Date.now() + this.delayMs;
+      }
+      this.promise.then(() => this.check());
+    }
```

Comment:
The Streams test were written before async/await were widely deployed, and we avoided using them for a while, but it's fine to use them now.

I suggest implementing this in terms of the `step_timeout` function provided by testharness.js, as it automatically adjusts for the slowness of the server when needed, and it already has a workaround for the builtin `setTimeout` being unavailable. Something like this:
```suggestion
    async check() {
      while (true) {
        await new Promise(resolve => step_timeout(resolve, this.delayMs));
        if (this.cancelled) {
          return;
        }
        this.callback();
      }
    }
```

Reply:
Sure, that seems fine!